pkg/network/probe_handler_test.go

```diff
@@ -96,8 +96,46 @@ func TestProbeHandlerSuccessfulProbe(t *testing.T) {
 				t.Errorf("prober.Do() = nil, expected an error")
 			}
 			if got != c.want {
-				t.Errorf("unexpected probe result: want: %t, got: %t", c.want, got)
+				t.Errorf("Unexpected probe result: want: %t, got: %t", c.want, got)
 			}
 		})
 	}
 }
+
+func BenchmarkProbeHandler(b *testing.B) {
+	var h http.Handler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Write([]byte("Inner Body"))
+	})
+	h = NewProbeHandler(h)
+	ts := httptest.NewServer(h)
```

Comment:
I don't think this test should need a "proper" server. It's only an HTTP handler after all and as such should be benchmarked in total isolation as it doesn't concern itself with talking to anything over i/o, so the test shouldn't either. The reported values will not be representative.